Lib/test/test_zip_threading.py

```diff
@@ -0,0 +1,42 @@
+import unittest
+from threading import Thread
+
+from test.support import threading_helper
+
+
+class ZipThreading(unittest.TestCase):
+    @staticmethod
+    def work(enum):
+        while True:
+            try:
+                next(enum)
+            except StopIteration:
+                break
+
+    @threading_helper.reap_threads
+    @threading_helper.requires_working_threading()
+    def test_threading(self):
+        number_of_threads = 8
+        number_of_iterations = 10
+        n = 40_000
+        enum = zip(range(n), range(n))
+        for _ in range(number_of_iterations):
+            worker_threads = []
+            for ii in range(number_of_threads):
+                worker_threads.append(
+                    Thread(
+                        target=self.work,
+                        args=[
+                            enum,
+                        ],
+                    )
+                )
+            _ = [t.start() for t in worker_threads]
+            _ = [t.join() for t in worker_threads]
+
+
+if __name__ == "__main__":
+    z = ZipThreading()
+    for kk in range(10):
+        print(kk)
+        z.test_threading()
```

Comment:
I guess there should be `unittest.main()`
Is this PR in a draft? I see the word `draft` in the title of the PR, but I'm not sure. If so, you can convert it to a draft by clicking `Convert to draft` button.

Additionally, I think it would make sense to move this file under the `Lib/test/test_free_threading/`.

Reply:
This PR is one of the options to resolve the issue. I will remove the draft label, although after discussion we might choose one of the other options.

And yes, `Lib/test/test_free_threading/` is a much better place for the test.